src/module/rules/rule-element/grant-item/base.ts

```diff
@@ -223,6 +209,37 @@ class GrantItemRuleElement extends RuleElementPF2e {
         }
     }
 
+    /** Set flags on granting and grantee items to indicate relationship between the two */
+    #setGrantFlags(granter: PreCreate<ItemSourcePF2e>, grantee: ItemSourcePF2e | ItemPF2e): void {
+        const flags = mergeObject(granter.flags ?? {}, { pf2e: { itemGrants: {} } });
+        flags.pf2e.itemGrants[this.flag] = {
+            // The granting item records the granted item's ID in an array at `flags.pf2e.itemGrants`
+            id: grantee instanceof ItemPF2e ? grantee.id : grantee._id,
+            // The on-delete action determines what will happen to the granter item when the granted item is deleted:
+            // Default to "detach" (do nothing).
+            onDelete: this.onDeleteActions?.grantee ?? "detach",
+        };
+
+        // The granted item records its granting item's ID at `flags.pf2e.grantedBy`
+        const grantedBy = {
+            id: granter._id,
+            // The on-delete action determines what will happen to the granted item when the granter is deleted:
+            // Default to "cascade" (delete the granted item) unless the granted item is physical.
+            onDelete:
+                this.onDeleteActions?.granter ?? setHasElement(PHYSICAL_ITEM_TYPES, grantee.type)
+                    ? "detach"
+                    : "cascade",
+        };
+
+        if (grantee instanceof ItemPF2e) {
+            // This is a previously granted item: update its grantedBy flag
+            // Don't await since it will trigger a data reset, possibly wiping temporary roll options
+            grantee.update({ "flags.pf2e.grantedBy": grantedBy }, { render: false });
```

Comment:
I forgot, does foundry filter prevent updates if the diff is empty?

Reply:
It doesn't make it to the server, but this is a grant new item with a randomly-generated ID.

Reply:
Then I guess even if somehow the logic ever tweaks so that this is used to "ensure" flags, it won't break from that.